pkg/node/node.go

```diff
@@ -101,105 +103,261 @@ func createAddressFilter(
 	}
 }
 
-// GenerateSignerForKeep generates a new threshold signer with ECDSA key pair. The
-// public key is a public key of the signing group. It publishes the public key
-// to the keep. It uses keep address as unique signing group identifier.
+// GenerateSignerForKeep generates a new threshold signer with ECDSA key pair
+// and submits the public key to the on-chain keep.
+//
+// The attempt for generating signer is retried on failure until the provided
+// context is done.
 func (n *Node) GenerateSignerForKeep(
 	ctx context.Context,
 	operatorPublicKey *operator.PublicKey,
 	keepAddress common.Address,
-	keepMembersIDs []tss.MemberID,
+	members []common.Address,
 ) (*tss.ThresholdSigner, error) {
 	memberID := tss.MemberIDFromPublicKey(operatorPublicKey)
 
-	signer, err := tss.GenerateThresholdSigner(
-		ctx,
-		keepAddress.Hex(),
-		memberID,
-		keepMembersIDs,
-		uint(len(keepMembersIDs)-1),
-		n.networkProvider,
-		n.tssParamsPool.get(),
-	)
-	if err != nil {
-		return nil, fmt.Errorf("failed to generate threshold signer: [%v]", err)
+	logger.Infof("Getting new key-generation pre-parameters from the pool")
+	preParamsBox := params.NewBox(n.tssParamsPool.get())
+	logger.Infof("Received new key-generation parameters from the pool")
```

Comment:
I hesitate whether at least one of these loggers should be changed to `debug` level.

Reply:
I am not sure, as an operator, I'd like to know how fast my node generates pre-parameters so that I can scale it up if needed. I probably don't want to enable debug level for that.

Reply:
We already got some loggers for that:
https://github.com/keep-network/keep-ecdsa/blob/3e4a0790b3f92453fd10ca950b480860a398fe78/pkg/node/params_pool.go#L45
https://github.com/keep-network/keep-ecdsa/blob/3e4a0790b3f92453fd10ca950b480860a398fe78/pkg/node/params_pool.go#L59-L63

Reply:
Oh, good point. In this case, I think we should drop those `Info` logs from `node.go` altogether. What do you think?

Reply:
I couldn't agree more!

Reply:
Done in https://github.com/keep-network/keep-ecdsa/pull/357/commits/e8b95aea03fd3e029a5ee5120110f90f9dfbdd5f